app/SavedLearningItem.php

```diff
@@ -0,0 +1,32 @@
+<?php
+
+namespace App;
+
+use Illuminate\Database\Eloquent\Model;
+
+class SavedLearningItem extends Model
+{
+    // Override the table used for the User Model
+    public $timestamps = false;
+
+    // Disable using created_at and updated_at columns
+
+    protected $table = 'saved_learning_items';
+
+    // Override the primary key column
+
+    protected $primaryKey = 'sli_id';
+
+    // Default
+    protected $fillable = [
+        'sli_id',
+        'category',
+        'item_id',
+        'student_id',
+    ];
+
+    public static function itemExists($id): bool
+    {
+        return SavedLearningItem::where('item_id', '=', $id)->count() > 0;
```

Comment:
Only checking for an ID means that if

**Item**
Category: tip
Id: 10

exists, and you try to save the following
**Item**
Category: activity (something other than tip)
Id: 10

We get true for `SavedLearningItem::itemExists`, but that combo `id`+`category` does not exist. So you'll have to check for `category` too in this query.

Reply:
This method would also be better suited for a `SavedLearningItemRepository`, instead of the in the Entity itself. Makes it easier to write a test because you can then easily mock the Repository.